File: api/analyze_video.js
// /api/analyze_video.js
const express = require('express');
const { generateLlmReport } = require('./_lib/helpers');

const app = express();

app.get('/api/analyze_video', async (req, res) => {
    const { id: videoId } = req.query;

    if (!videoId) {
        return res.status(400).type('text/plain').send("Error: A video 'id' parameter is required.");
    }

    const report = await generateLlmReport(videoId);
    
    res.setHeader('Content-Type', 'text/plain; charset=utf-8');
    return res.status(200).send(report);
});

module.exports = app;

File: api/analyze_video.py
# api/analyze_video.py

import asyncio
import aiohttp
from flask import Flask, request, Response

# Import the main helper function
from .llm_handler import generate_llm_report

# This file also gets its own Flask app instance
app = Flask(__name__)

@app.route('/', defaults={'path': ''})
@app.route('/<path:path>')
async def handler(path):
    # This is the exact same logic from your old analyze_video_for_llm function
    video_id = request.args.get('id')
    if not video_id:
        return jsonify({"error": "A video 'id' parameter is required."}), 400

    async with aiohttp.ClientSession() as session:
        report = await generate_llm_report(session, video_id)
    
    return Response(report, mimetype='text/plain; charset=utf-8')
